keras_cv/models/object_detection/yolo_v8/yolo_v8.py

```diff
@@ -0,0 +1,604 @@
+    inputs,
+    output_channel,
+    kernel_size=1,
+    strides=1,
+    activation="swish",
+    name=None,
+):
+    if kernel_size > 1:
+        inputs = layers.ZeroPadding2D(
+            padding=kernel_size // 2, name=f"{name}_pad"
+        )(inputs)
+
+    nn = layers.Conv2D(
+        filters=output_channel,
+        kernel_size=kernel_size,
+        strides=strides,
+        padding="valid",
+        use_bias=False,
+        name=f"{name}_conv",
+    )(inputs)
+    nn = layers.BatchNormalization(
+        momentum=BATCH_NORM_MOMENTUM,
+        epsilon=BATCH_NORM_EPSILON,
+        name=f"{name}_bn",
+    )(nn)
+    nn = layers.Activation(activation, name=name)(nn)
+    return nn
+
+
+def csp_with_2_conv(
+    inputs,
+    channels=-1,
+    depth=2,
+    shortcut=True,
+    expansion=0.5,
+    activation="swish",
+    name=None,
+):
+    channel_axis = -1
+    channels = channels if channels > 0 else inputs.shape[channel_axis]
+    hidden_channels = int(channels * expansion)
+
+    pre = conv_bn(
+        inputs,
+        hidden_channels * 2,
+        kernel_size=1,
+        activation=activation,
+        name=f"{name}_pre",
+    )
+    short, deep = tf.split(pre, 2, axis=channel_axis)
+
+    out = [short, deep]
+    for id in range(depth):
+        deep = conv_bn(
+            deep,
+            hidden_channels,
+            kernel_size=3,
+            activation=activation,
+            name=f"{name}_pre_{id}_1",
+        )
+        deep = conv_bn(
+            deep,
+            hidden_channels,
+            kernel_size=3,
+            activation=activation,
+            name=f"{name}_pre_{id}_2",
+        )
+        deep = (out[-1] + deep) if shortcut else deep
+        out.append(deep)
+    out = tf.concat(out, axis=channel_axis)
+    out = conv_bn(
+        out,
+        channels,
+        kernel_size=1,
+        activation=activation,
+        name=f"{name}_output",
+    )
+    return out
+
+
+def spatial_pyramid_pooling_fast(
+    inputs, pool_size=5, activation="swish", name=None
+):
+    channel_axis = -1
+    input_channels = inputs.shape[channel_axis]
+    hidden_channels = int(input_channels // 2)
+
+    nn = conv_bn(
+        inputs,
+        hidden_channels,
+        kernel_size=1,
+        activation=activation,
+        name=f"{name}_pre",
+    )
+    pool_1 = layers.MaxPool2D(
+        pool_size=pool_size, strides=1, padding="same", name=f"{name}_pool1"
+    )(nn)
+    pool_2 = layers.MaxPool2D(
+        pool_size=pool_size, strides=1, padding="same", name=f"{name}_pool2"
+    )(pool_1)
+    pool_3 = layers.MaxPool2D(
+        pool_size=pool_size, strides=1, padding="same", name=f"{name}_pool3"
+    )(pool_2)
+
+    out = tf.concat([nn, pool_1, pool_2, pool_3], axis=channel_axis)
+    out = conv_bn(
+        out,
+        input_channels,
+        kernel_size=1,
+        activation=activation,
+        name=f"{name}_output",
+    )
+    return out
+
+
+# This should probably just be a CSPDarknet. That's an outstanding TODO
```

Comment:
Ah in that case my nits on the code above are less relevant

Reply:
Do you think we could load their weights into our CSPDarknetBackbone? Or are the architectures not canonical?

Reply:
I think we probably need some modifications to our CSPDarknetBackbone to support this (namely, right now ours seems to always have 5 stacks? and many of these have only 4).

I'm not certain. It could be that it's not really the same architecture exactly